app/scripts/controllers/transactions/index.js

```diff
@@ -184,13 +186,29 @@ class TransactionController extends EventEmitter {
   @returns {txMeta}
   */
 
-  async addUnapprovedTransaction (txParams) {
+  async addUnapprovedTransaction (txParams, origin) {
+
     // validate
     const normalizedTxParams = txUtils.normalizeTxParams(txParams)
-    // Assert the from address is the selected address
-    if (normalizedTxParams.from !== this.getSelectedAddress()) {
-      throw new Error(`Transaction from address isn't valid for this account`)
+
+    // TODO:lps once, I saw 'origin' with a value of 'chrome-extension://...'
+    // for a transaction initiated from the UI.
+    // When would this happen? It's a problem if it does. I don't think we want to
+    // hard-code 'chrome-extension://' here
```

Comment:
not sure this is a hardcode thing it might be more how the provider is setup i think we used to check origin and if it was the extensions origin replace it with metamask this must have changed if this is happening? will find code to confirm

Reply:
correction we check host name and then make origin = hostname

Reply:
Mark did some work with chrome extension IDs and origins. I don't know if it affects the one time I saw `chrome-extension://` there, but it may be safe to ignore?

Reply:
maybe i wouldnt outright ignore it

Reply:
was this on your own machine?

Reply:
was it in fulpage mode?

Reply:
I'm not sure about the specific conditions. The good news is that it will throw an error if the origin is something unrecognized, I pushed a change attaching the origin to the error thrown.

Reply:
The recent changes I made were to allow external extensions to connect to us like a dapp would. I'm having difficulty imaging how that could be related to the scenario you're describing, where a transaction from our UI was misidentified as _not_ being internal. External extensions don't even use the same extension connect handler - they message via `onConnectExternal` instead of `onConnect`.

I'll keep an eye out for anything that might be related though. Certainly let us know if you encounter that again. It sounds like a pretty serious problem if it is still occurring.

Reply:
Yeah, I only saw it once, and I have no idea what my local state was, but the, err, good news is that everything will fail as opposed to let something unpermitted succeed. Ideally, I'd like to put these checks to the permissions middleware so that things blow up even before they touch the transactions controller. The transaction controller shouldn't have to be aware of permissions or selected addresses, IMO.

Reply:
@frankiebee , is there any way for a transaction to be submitted without passing through the usual RpcEngine middleware stack?

Reply:
yes two the edit but then you would be hand filling the origin param so it would be easy to find